### ISSUE TITLE
exhibition space fixes

### ISSUE THREAD

**Issue by foxm** [blakearchive/archive#335]
@queryluke @joeafletch guys, i finally added the exhibition space that joe v. has wanted for a long time. there's a menu item for it, "Archive Exhibits." the first one is joe's essay on Illuminated Printing. have a look. you'll notice it opens up into the Reading Mode initially. luke, there are a couple issues with it that i haven't been able to resolve. would you mind having checking them out if you have time. i imagine they require easy fixes:

1) menu items in the text section of the exhibit, when clicked, need to scroll within that div, not within the whole browser
2) the captions, which are technically transcriptions in the BAD for this exhibit (illuminatedprintingexhibit.xml) need to wrap within a fixed width column next to their corresponding images
3) it would be nice to link the superscript in-text citations (ill 1., ill 2., etc.) to the corresponding image in the gallery so that clicking the superscript would scroll the gallery to the corresponding image.
4) when in Gallery Mode for an exhibit, the rotate feature doesn't resize the object view window when the length of the image is greater than the width. i'm guessing this is a problem elsewhere. it used to work. it must have been broken somewhere along the way.
5) i need to add enlargement images so that the Enlargement button works
6) i'll probably take out the Generate Citation and implement that after the launch

**Comment by queryluke** [comment]
@foxm 

Mike, honestly I'm very concerned about these changes.

First, this functionality was never on the table. So I can't justify spending time debugging it.

But what really concerns me is the data integrity. Essentially we're now munging blake object/copies/works (of which the interconnectedness is complicated enough) with a completely new data type "exhibit", which isn't even a blake work. This has major implications for the data structure AND ui. We can't continue to mash functionality together with if logic based on work types, virtualness, etc.

While I commend your initiative for using the tools available to achieve this idea, I think we need to take a step back and consider this a brand new site feature. Something that can be addressed in a future release. Or translated into a static page or some kind.

Either way, we'll need to discuss this in the next meeting.

**Comment by foxm** [comment]
@queryluke 

luke, actually this has been on the table since before i got here three years ago. and yes, it uses the existing technology. an exhibit is a stripped down work with an essay attached to it. the way i implemented it avoids us having to modify the import code and to think of exhibits as different things. and i've made a new controller, exhibitController, to handle the display. there is some if logic, as you point out, in the reading view and the object view. we can separate that out, making different directives for the exhibits that would be very similar to the directives for the works. but i want to continue handling exhibits like stripped down works. it makes publishing them very easy.

by the way, the example exhibit i've done is treated as a static page with extra functionality on the current site:

http://www.blakearchive.org/exist/blake/archive/biography.xq?b=illum&targ_div=d1

**Comment by foxm** [comment]
@queryluke luke, if you want, you can branch off the exhibition space code to its own branch. is that possible? and then you could roll back master to commit 3f1a08614ca8c42c4dcdff5e32f6fa28ecef50e7 

i'd do it, but i've never done so before, and i don't want to screw anything up.

**Comment by queryluke** [comment]
I don't think we need to roll anything back. We can work from what's already there and suppress links if need be.  But exhibits should be considered it's own structure, meaning they should have their own import scripts, views, controllers, logic, etc. It's something we'll discuss in wednesday's meeting.

**Comment by foxm** [comment]
ok. but i'm not sure they should get their own structure. we've already had much discussion about the exhibition space. joe v. will update you on wednesday.
